### PR TITLE
CEDS-5413 Section 2 Pages

### PR DESCRIPTION
**Pull request by Dweik97**

### CODE REVIEW

File: src/test/scala/uk/gov/hmrc/test/ui/pages/section2/AreYouTheExporterPage.scala [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  val title: String          = "Are you the exporter?"
+  def backButtonHref: String = detail(DeclarationType) match {
+    case Clearance => DeclarantDetailsPage.path
+    case _         => SummarySection1Page.path
+  }
+
+  override val expanderHrefs: Map[String, Seq[String]] =
+    Map(Common -> List(areYouTheExporter), Clearance -> List(areYouTheExporterCL))
+
+  override protected def fillPage(values: String*): Unit = {
```
Comment (dhanist-gadhiraju) [inline src/test/scala/uk/gov/hmrc/test/ui/pages/section2/AreYouTheExporterPage.scala:38]: can you remove protected for all the methods as these methods are called in stepdefs

File: src/test/scala/uk/gov/hmrc/test/ui/pages/section2/AuthorisationChoicePage.scala [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+    Map(
+      Common    -> authorisationChoiceTariffLinks,
+      Clearance -> authorisationChoiceTariffLinks
+    )
+  }
+
+  val exportProcedure = 0
+
+  override protected def fillPage(values: String*): Unit =
```
Comment (dhanist-gadhiraju) [inline src/test/scala/uk/gov/hmrc/test/ui/pages/section2/AuthorisationChoicePage.scala:48]: you are not storing the value in the cache. we need this to be stored

File: src/test/scala/uk/gov/hmrc/test/ui/pages/section2/AuthorisationsRequiredPage.scala [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    }
+  val backButtonHref: String = AddAuthorisationRequiredPage.path
+
+  val action          = 0
+  val authToBeRemoved = 1
+  val yesNoConfirm    = 2
+
+  override protected def fillPage(values: String*): Unit =
+    values(action) match {
+      case "Remove" => AuthorisationRemovePage.fillPage(values(authToBeRemoved), values(yesNoConfirm))
```
Comment (dhanist-gadhiraju) [inline src/test/scala/uk/gov/hmrc/test/ui/pages/section2/AuthorisationsRequiredPage.scala:38]: Authorisation removed should is a separate page right. Please can you explain why do you we need remove

File: src/test/scala/uk/gov/hmrc/test/ui/pages/section2/CarrierEORINumberPage.scala [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  val title: String          = "Do you know the EORI number of your carrier or haulier?"
+  def backButtonHref: String = maybeDetail(ExporterEORI) match {
+    case Some(_) => AreYouTheExporterPage.path
+    case None    => RepresentationTypeAgreedPage.path
+  }
+
+  override val expanderHrefs: Map[String, Seq[String]] =
+    Map(Common -> List(carrierEoriNumber), Clearance -> List(carrierEoriNumberCL))
+
+  override protected def fillPage(values: String*): Unit = {
```
Comment (dhanist-gadhiraju) [inline src/test/scala/uk/gov/hmrc/test/ui/pages/section2/CarrierEORINumberPage.scala:36]: can you specify what values we are passing as a comment

File: src/test/scala/uk/gov/hmrc/test/ui/pages/section2/AreYouTheExporterPage.scala (hunk 1 above)
Comment (Dweik97) [reply src/test/scala/uk/gov/hmrc/test/ui/pages/section2/AreYouTheExporterPage.scala:38]: Could've sworn I did that, I don't know why they're back 😕. I will update them.

File: src/test/scala/uk/gov/hmrc/test/ui/pages/section2/IsAuthorisationRequiredPage.scala [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+  val path: String           = "/declaration/is-authorisation-required"
+  val title: String          = "Do you have any authorisations you want to declare?"
+  val backButtonHref: String = AuthorisationChoicePage.path
+
+  override val expanderHrefs: Map[String, Seq[String]] = Map(
+    Common    -> List(isAuthorisationRequired, isAuthorisationRequired1, isAuthorisationRequired2),
+    Clearance -> List(isAuthorisationRequiredCL, isAuthorisationRequiredCL1, isAuthorisationRequiredCL2)
+  )
+
+  override protected def fillPage(values: String*): Unit =
```
Comment (siva1982) [inline src/test/scala/uk/gov/hmrc/test/ui/pages/section2/IsAuthorisationRequiredPage.scala:35]: please take out protected one

**Review by udinobi — APPROVED** [review]

**Pull request merged**